test/controllers/trace_controller_test.rb

```diff
@@ -389,115 +394,115 @@ def test_data_not_found
+    get :picture, :display_name => public_trace_file.user.display_name, :id => public_trace_file.id
     check_trace_picture public_trace_file
 
     # Now with some other user, which should work since the trace is public
-    get :picture, { :display_name => users(:normal_user).display_name, :id => public_trace_file.id }, { :user => users(:public_user) }
+    get :picture, { :display_name => public_trace_file.user.display_name, :id => public_trace_file.id }, { :user => create(:user) }
     check_trace_picture public_trace_file
 
     # And finally we should be able to do it with the owner of the trace
-    get :picture, { :display_name => users(:normal_user).display_name, :id => public_trace_file.id }, { :user => users(:normal_user) }
+    get :picture, { :display_name => public_trace_file.user.display_name, :id => public_trace_file.id }, { :user => public_trace_file.user }
     check_trace_picture public_trace_file
   end
 
   # Check the picture for an anonymous trace can't be downloaded by another user
   def test_picture_anon
-    anon_trace_file = create(:trace, :visibility => "private", :user => users(:public_user), :fixture => "b")
+    anon_trace_file = create(:trace, :visibility => "private", :fixture => "b")
 
     # First with no auth
-    get :picture, :display_name => users(:public_user).display_name, :id => anon_trace_file.id
+    get :picture, :display_name => anon_trace_file.user.display_name, :id => anon_trace_file.id
     assert_response :forbidden
 
     # Now with some other user, which shouldn't work since the trace is anon
-    get :picture, { :display_name => users(:public_user).display_name, :id => anon_trace_file.id }, { :user => users(:normal_user) }
+    get :picture, { :display_name => anon_trace_file.user.display_name, :id => anon_trace_file.id }, { :user => create(:user) }
     assert_response :forbidden
 
     # And finally we should be able to do it with the owner of the trace
-    get :picture, { :display_name => users(:public_user).display_name, :id => anon_trace_file.id }, { :user => users(:public_user) }
+    get :picture, { :display_name => anon_trace_file.user.display_name, :id => anon_trace_file.id }, { :user => anon_trace_file.user }
     check_trace_picture anon_trace_file
   end
 
   # Test downloading the picture for a trace that doesn't exist
   def test_picture_not_found
     # First with no auth, which should work since the trace is public
-    get :picture, :display_name => users(:public_user).display_name, :id => 0
+    get :picture, :display_name => create(:user).display_name, :id => 0
     assert_response :not_found
 
     # Now with some other user, which should work since the trace is public
-    get :picture, { :display_name => users(:public_user).display_name, :id => 0 }, { :user => users(:public_user) }
+    get :picture, { :display_name => create(:user).display_name, :id => 0 }, { :user => create(:user) }
```

Comment:
That's not doing the same thing any more as it is now authenticating as a different user to the one that nominally owns the non-existent trace.

In the previous test that did the same thing you created the deleted trace up front and used that user throughout.